packages/plugin-react/src/index.d.ts

```diff
@@ -1,12 +1,29 @@
 import { Plugin } from 'vite'
-import { ParserOptions } from '@babel/core'
-
-type PluginFactory = (options?: Options) => Plugin
-
-declare const createPlugin: PluginFactory & { preambleCode: string }
+import { TransformOptions, ParserOptions } from '@babel/core'
 
 export interface Options {
-  parserPlugins: ParserOptions['plugins']
+  /**
+   * Enable `react-refresh` integration. Vite disables this in prod env or build mode.
+   * @default true
+   */
+  fastRefresh?: boolean
+  /**
+   * Set this to `"automatic"` to use [vite-react-jsx](https://github.com/alloc/vite-react-jsx).
+   * @default "classic"
+   */
+  jsxRuntime?: 'classic' | 'automatic'
+  /**
+   * Babel configuration applied in both dev and prod.
+   */
+  babel?: TransformOptions
+  /**
+   * @deprecated Use `babel.parserOpts.plugins` instead
+   */
+  parserPlugins?: ParserOptions['plugins']
 }
 
-export default createPlugin
+declare const viteReact: (options?: Options) => Plugin
+
+export default viteReact
+
+export const preambleCode: string
```

Comment:
Oops, I need to actually export this part. Is it exported for middleware mode, or what?